include/model/CTokenListCategory.h

```diff
@@ -99,11 +99,35 @@ class MODEL_EXPORT CTokenListCategory {
     //!     if (category.missingCommonTokenWeight(uniqueTokenIds) == 0)
     //! instead of calling this method.  However, this method is much faster
     //! as it can return false as soon as a mismatch occurs.
-    bool isMissingCommonTokenWeightZero(const TSizeSizeMap& uniqueTokenIds) const;
+    template<typename PAIR_CONTAINER>
+    bool isMissingCommonTokenWeightZero(const PAIR_CONTAINER& uniqueTokenIds) const {
+
+        auto commonIter = m_CommonUniqueTokenIds.begin();
+        auto testIter = uniqueTokenIds.begin();
+        while (commonIter != m_CommonUniqueTokenIds.end() &&
+               testIter != uniqueTokenIds.end()) {
+            if (commonIter->first < testIter->first) {
+                return false;
+            }
```

Comment:
I more-or-less did this. The second part of the condition has to be `testIter->first != commonIter->first || testIter->second != commonIter->second` instead of simply `*testIter != *commonIter` because in some cases one is a `std::pair<const std::size_t, std::size_t>` and the other is a `std::pair<std::size_t, std::size_t>`.

Reply:
Ok, great. It's a shame comparison doesn't work with const. That feels like an error in the std::pair implementation.